test/k8s-integration/main.go

```diff
@@ -424,27 +422,23 @@ func generateGCETestSkip(testParams *testParameters) string {
 func generateGKETestSkip(testParams *testParameters) string {
 	skipString := "\\[Disruptive\\]|\\[Serial\\]"
 	curVer := mustParseVersion(testParams.clusterVersion)
-	// "volumeMode should not mount / map unused volumes in a pod" tests a
-	// (https://github.com/kubernetes/kubernetes/pull/81163)
-	// bug-fix introduced in 1.16
-	if curVer.lessThan(mustParseVersion("1.16.0")) {
-		skipString = skipString + "|volumeMode\\sshould\\snot\\smount\\s/\\smap\\sunused\\svolumes\\sin\\sa\\spod"
-	}
-
-	// For GKE deployed PD CSI driver, resizer sidecar is enabled in 1.16.8-gke.3
-	if curVer.lessThan(mustParseVersion("1.16.0")) {
-		skipString = skipString + "|allowExpansion"
+	var nodeVer *version
+	if testParams.nodeVersion != "" {
+		nodeVer = mustParseVersion(testParams.nodeVersion)
 	}
 
 	// Skip fsgroup change policy based on GKE cluster version.
-	if curVer.lessThan(mustParseVersion("1.20.0")) {
+	if curVer.lessThan(mustParseVersion("1.20.0")) || (nodeVer != nil && nodeVer.lessThan(mustParseVersion("1.20.0"))) {
 		skipString = skipString + "|fsgroupchangepolicy"
 	}
 
 	// Running snapshot tests on GKE needs k8s 1.19x storage e2e testsuite (until GKE supports v1 snapshot APIs).
```

Comment:
Running snapshot tests on GKE clusters which does not support v1 snapshot CRDs needs 1.19x storage e2e testsuites...